From 7ad77ff712e541a766f7b19475537c3e8b07350b Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Fri, 17 Oct 2014 19:44:39 -0400
Subject: [scsi] cxgb4i: Remove duplicate call to dst_neigh_lookup()

Message-id: <1413575081-32047-5-git-send-email-svemuri@redhat.com>
Patchwork-id: 97657
O-Subject: [RHEL7.1 PATCH BZ 1153834 4/6] cxgb4i: Remove duplicate call to dst_neigh_lookup()
Bugzilla: 1153834
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Upstream : https://git.kernel.org/cgit/linux/kernel/git/davem/net.git/commit/?id=c5bbcb5822b25c9f738db98e6d6ad2506cab8136

There is an extra call to dst_neigh_lookup() leftover in cxgb4i that can cause
an unreleased refcnt issue. Remove extraneous call.

Signed-off-by: Anish Bhatt <anish@chelsio.com>

Fixes : 759a0cc5a3e1b ('cxgb4i: Add ipv6 code to driver, call into libcxgbi ipv6 api')
Signed-off-by: David S. Miller <davem@davemloft.net>

(cherry picked from commit c5bbcb5822b25c9f738db98e6d6ad2506cab8136)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c b/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c
index df176f0..8c3003b 100644
--- a/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c
+++ b/drivers/scsi/cxgbi/cxgb4i/cxgb4i.c
@@ -1317,11 +1317,6 @@ static int init_act_open(struct cxgbi_sock *csk)
  cxgbi_sock_set_flag(csk, CTPF_HAS_ATID);
  cxgbi_sock_get(csk);
 
- n = dst_neigh_lookup(csk->dst, &csk->daddr.sin_addr.s_addr);
- if (!n) {
-  pr_err("%s, can't get neighbour of csk->dst.\n", ndev->name);
-  goto rel_resource;
- }
  csk->l2t = cxgb4_l2t_get(lldi->l2t, n, ndev, 0);
  if (!csk->l2t) {
   pr_err("%s, cannot alloc l2t.\n", ndev->name);
-- 
1.7.1